community/modules/network/rdma-vpc/main.tf

```diff
@@ -125,6 +125,23 @@ locals {
       alias_ip_range     = []
     }
   ]
+
+  # FIX_ME(arajmane): There is a concern about this not working in a shared VPC environment. 
+  # To unblock experimental testing, we decided to go ahead with this.
```

Comment:
I believe that this choice (`subnetwork_project = null`) will run into problems in any environment that does not have a provider set at the project level.

It should not be merged as-is unless you can demonstrate that adding `subnetwork_project = var.project_id` causes known problems.

Reply:
Specifically your code is going to error when there is no default project specified in the provider. It's better not to fall back to that.

It will hit this:

https://github.com/hashicorp/terraform-provider-google/blob/main/google/tpgresource/field_helpers.go#L397

after hitting this:

https://github.com/hashicorp/terraform-provider-google/blob/00a52318354fe60774305a53606a7c2f7ff6a9d3/google/services/compute/compute_instance_helpers.go#L466

Reply:
@tpdownes yes, will update the `subnetwork_project = var.project_id`. 

Ideally, I understand we should be using self_link. But, looks like the GKE sandbox environment limitation to me, ATM. 

That said, I am going to check with the GKE team to see why in the sandbox environment self_link isn't working for subnetwork. In fact, similarly when creating the `system_node_pools` we are needing to pass the cluster name instead of self link.

Reply:
see modules/scheduler/gke-cluster/main.tf in this PR